### DISCUSSION TITLE
Different External IP

### DISCUSSION THREAD

**Discussion by bjkeller7** [rejetto/hfs#755] [General]
Hello rejetto,

 I have a domain setup to a different external IP address i own and have the port forwarding setup correctly but it will not connect. I know the port forwarding is correct becuase if I change it to look at my main WAN IP it works.  Is there a way to change what external IP address the server is looking at to "verify"?

I'm using version 0.53.1

Thanks

**Comment by rejetto** [comment]
hfs will first verify using the address you configured as "main address"
<img width="162" alt="image" src="https://github.com/user-attachments/assets/4674d733-06e8-48cd-82b3-6182e2004d14">

**Reply by bjkeller7** [reply]
![Screenshot 2024-10-03 062654](https://github.com/user-attachments/assets/7b143bac-5cf6-42fb-8ef7-e768dd77305a)
I have that set but this still trying to verify off my WANIP

**Comment by rejetto** [comment]
the check on the domain is warning you that the IP is not the expected one, but there's a "continue" button.
Did you continue?
the test will continue using the domain.

needless to say, the "verify" function is not necessary for your hfs to work, it's just a tool.

**Comment by bjkeller7** [comment]
Ok, thank you.  I switched it over to a different IP I have and it started work.  Thanks for the help.

**Comment by noagenda33** [comment]
How do I change my WAN IP? I have 5, but it's defaulting to the first one. Needs to be "xxx.178.38.103"
<img width="761" height="282" alt="2025-12-26_07-47-32 (1)" src="https://github.com/user-attachments/assets/4554e577-a065-44b3-b4b4-1df54fb1f65e" />

**Reply by rejetto** [reply]
there's no way to change that. HFS will just show all the public IPs it can find.
is the other IP (103) an internet/public IP?
anyway, you can set any IP clicking "change"  after "main address". That should apply to the "verify" link. 
just to be clear, you don't really *need* to have the right IP there, because that's just to be used for the "verify", which is a tool absolutely not necessary for HFS to work.

**Comment by noagenda33** [comment]
My Public IP scope is 101-105. I am using 103 for HFS. It doesn't automatically see that. I have put my FQDN in the "change" address box. It all seems to work, but I don't understand why I can't change it in the screenshot above.

**Reply by rejetto** [reply]
well, it's a mixed situation. That's mostly to show what HFS has found, and for "showing" there's no point in letting the user entering it.
But you also have the "verify" feature right there. I guess that's what concerns you, right?
It's not clear that the *verify* will act first on the "main address". This can be improved, but i'm not sure how. Maybe moving the link/button somewhere else. I can't see a perfect solution.

The best would be to find all your IPs. HFS relies on external services for this. For IPv4 it is currently using these:
```json
"http://ipv4.icanhazip.com",
{
    "v": 4, "type": "dns",
    "ips": ["208.67.222.222", "208.67.220.220"],
    "name": "myip.opendns.com", "dnsRecord": "A"
},
{
    "v": 4, "type": "dns",
    "ips": ["216.239.32.10", "216.239.34.10", "216.239.36.10", "216.239.38.10"],
    "name": "o-o.myaddr.l.google.com", "dnsRecord": "TXT"
},
```

HFS  tries all of them and shows all results. This means that in your case all of them are returning only that single IP. I'm not aware of any way to get more, but if you are, and you think it can be used by HFS, I'd like to know it.
I've never met another case like yours, with 5 addresses.

**Comment by noagenda33** [comment]
My ISP offers static IP address packages. I purchased 5. I plug them into my Unifi Dream Machine and can assign them to hosts on my network. Port forward, etc. Then I create an Host A record at my Domain Registrar using one of the ISP assign Static IP address's.
<img width="1355" height="1210" alt="2025-12-27_05-15-45" src="https://github.com/user-attachments/assets/848226d6-99cd-4565-813c-8a508890180b" />

<img width="957" height="299" alt="2025-12-27_05-25-20" src="https://github.com/user-attachments/assets/a588d23f-2680-43af-810c-98be127df511" />

**Reply by rejetto** [reply]
thanks for sharing, your case is interesting, but not very common, i guess.
It seems that your machine .50 is not going out with the .103, but the rule is only working when receiving connections.
Here is what GPT has to say about it:

This is **not an HFS issue**. It’s normal behavior when multiple public IPs are assigned to the same WAN.

### What’s happening
- The WAN has multiple public IPs (e.g. `.101–.105`)
- HFS is exposed via port forwarding on `.103`
- When HFS checks its public IP, it reports `.101`

### Why
- **Port forwarding affects inbound traffic only**
- **Outbound traffic uses the WAN’s primary/default IP**
- UniFi does not automatically bind outbound traffic to the forwarded IP

Flow:
HFS → outbound request  
Router → SNAT using `.101`  
IP-check service → reports `.101`

### Correct fix
Add a **Source NAT (SNAT)** rule in UniFi:
- **Source**: HFS LAN IP  
- **Outbound interface**: WAN1  
- **Translation address**: public IP `.103`

### Key rule
Port forwarding controls inbound traffic.  
Source NAT controls the outbound public IP.

**Comment by noagenda33** [comment]
Wow, that totally worked. Learned something new. Thanks.
<img width="791" height="603" alt="2025-12-27_17-12-18" src="https://github.com/user-attachments/assets/e50c6055-0ead-4cc4-a54f-77921f0e77ce" />
